src/wirecraft/client/__init__.py

```diff
@@ -7,154 +7,253 @@
 from .assets import SWITCH_DEVICE
 from .server_interface import ServerInterface
 
-FPS = pygame.time.Clock()
-FPS.tick(30)
 BLACK = (0, 0, 0)
+WHITE = (255, 255, 255)
 RES_LIST = [(3024, 1964), (1920, 1080), (1900, 1200)]
 FLAGS = pygame.FULLSCREEN | pygame.NOFRAME | pygame.SCALED
+FPS = 30
 
 
 @dataclass
 class Camera:
-    x: int
-    y: int
+    x: float
+    y: float
     zoom: float
+    min_zoom: float = 0.3
+    max_zoom: float = 3.0
+
+    def screen_to_world(self, screen_pos: tuple[float, float], screen_size: tuple[int, int]) -> tuple[float, float]:
+        """Convert screen coordinates to world coordinates
+        screen coordinates are relative to the top left corner of the screen, world coordinates are relative to the center of the screen
+
+        Args:
+            screen_pos (tuple[float, float]): the position on the screen
+            screen_size (tuple[int, int]): the size of the screen
+
+        Returns:
+            tuple[float, float]: the position in the world
```

Comment:
```suggestion
            screen_pos: the position on the screen
            screen_size: the size of the screen

        Returns:
            the position in the world
```

Remove types as they are defined in the function signature.